pkg/registry/apps/statefulset/strategy_test.go

```diff
@@ -204,3 +209,81 @@ func TestStatefulSetStatusStrategy(t *testing.T) {
 		t.Errorf("unexpected error %v", errs)
 	}
 }
+
+func makeStatefulSetWithMaxUnavailable(maxUnavailable *int) *apps.StatefulSet {
+	rollingUpdate := apps.RollingUpdateStatefulSetStrategy{}
+	if maxUnavailable != nil {
+		maxUnavailableIntStr := intstr.FromInt(*maxUnavailable)
+		rollingUpdate = apps.RollingUpdateStatefulSetStrategy{
+			MaxUnavailable: &maxUnavailableIntStr,
+		}
+	}
+
+	return &apps.StatefulSet{
+		Spec: apps.StatefulSetSpec{
+			UpdateStrategy: apps.StatefulSetUpdateStrategy{
+				Type:          apps.RollingUpdateStatefulSetStrategyType,
+				RollingUpdate: &rollingUpdate,
+			},
+		},
+	}
+}
+
+func TestDropDisabledField(t *testing.T) {
+	maxUnavailable := 3
+	testCases := []struct {
+		name                 string
+		enableMaxUnavailable bool
+		sts                  *apps.StatefulSet
+		oldSts               *apps.StatefulSet
+		expectedSts          *apps.StatefulSet
+	}{
+		{
+			name:                 "MaxUnavailable not enabled, field not used",
+			enableMaxUnavailable: false,
+			sts:                  makeStatefulSetWithMaxUnavailable(nil),
+			oldSts:               nil,
+			expectedSts:          makeStatefulSetWithMaxUnavailable(nil),
+		},
+		{
+			name:                 "MaxUnavailable not enabled, field used in new, not in old",
+			enableMaxUnavailable: false,
+			sts:                  makeStatefulSetWithMaxUnavailable(&maxUnavailable),
+			oldSts:               nil,
+			expectedSts:          makeStatefulSetWithMaxUnavailable(nil),
+		},
+		{
+			name:                 "MaxUnavailable not enabled, field used in old and new",
+			enableMaxUnavailable: false,
+			sts:                  makeStatefulSetWithMaxUnavailable(&maxUnavailable),
+			oldSts:               makeStatefulSetWithMaxUnavailable(&maxUnavailable),
+			expectedSts:          makeStatefulSetWithMaxUnavailable(&maxUnavailable),
+		},
+		{
+			name:                 "MaxUnavailable enabled, field used",
+			enableMaxUnavailable: true,
+			sts:                  makeStatefulSetWithMaxUnavailable(&maxUnavailable),
+			oldSts:               nil,
+			expectedSts:          makeStatefulSetWithMaxUnavailable(&maxUnavailable),
+		},
+
+		/* add more tests for other dropped fields as needed */
```

Comment:
Can you add a test case for maxUnavailable enabled, field used in old and new

Reply:
done

Reply:
remove the comment.